src/passes/OptimizeInstructions.cpp

```diff
@@ -1272,15 +1284,26 @@ struct OptimizeInstructions
       return right;
     }
     // x == 0   ==>   eqz x
-    if ((matches(curr, binary(Abstract::Eq, any(&left), ival(0))))) {
-      return builder.makeUnary(EqZInt64, left);
+    if (matches(curr, binary(Abstract::Eq, any(&left), ival(0)))) {
+      return builder.makeUnary(Abstract::getUnary(left->type, Abstract::EqZ),
+                               left);
     }
     // Operations on one
     // (signed)x % 1   ==>   0
     if (matches(curr, binary(Abstract::RemS, pure(&left), ival(1)))) {
       right->value = Literal::makeSingleZero(type);
       return right;
     }
+    // (signed)x % C_pot != 0   ==>  x & (C_pot - 1) != 0
+    if (matches(curr,
+                binary(Abstract::Ne,
+                       binary(&inner, Abstract::RemS, any(&left), constant(&c)),
+                       ival(0)))) {
+      if (IsPowerOf2((uint64_t)c->value.getInteger())) {
```

Comment:
It would be good to `&&` this condition with the previous one to reduce the amount of indentation necessary.

Reply:
Done!